Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapFragment.java

```diff
@@ -0,0 +1,928 @@
+        for (LatLong coord : footprintToBeDraw.getVertexInGlobalFrame()) {
+            pathPoints.add(CoordToLatLang(coord));
+        }
+        pathOptions.points(pathPoints);
+
+        map.addOverlay(pathOptions);
+
+    }
+
+    /**
+     * Save the map camera state on a preference file
+     * http://stackoverflow.com/questions
+     * /16697891/google-maps-android-api-v2-restoring
+     * -map-state/16698624#16698624
+     */
+    @Override
+    public void saveCameraPosition() {
+        final BaiduMap map = getBaiduMap();
+		if (map == null) return;
+
+		final MapStatus mapStatus = map.getMapStatus();
+        mAppPrefs.prefs.edit()
+                .putFloat(PREF_LAT, (float) mapStatus.target.latitude)
+                .putFloat(PREF_LNG, (float) mapStatus.target.longitude)
+                .putFloat(PREF_BEA,  mapStatus.rotate)   //FIXME: is bearing equal to rotate in baidu map?
+                .putFloat(PREF_TILT, mapStatus.overlook) //FIXME: is tilt equal to overlook in baidu map?
+                .putFloat(PREF_ZOOM, mapStatus.zoom).apply();
+    }
+
+    @Override
+    public void loadCameraPosition() {
+        final BaiduMap map = getBaiduMap();
+		if (map == null) return;
+
+		final SharedPreferences settings = mAppPrefs.prefs;
+		final MapStatus.Builder camera = new MapStatus.Builder();
+
+		camera.rotate(settings.getFloat(PREF_BEA, DEFAULT_BEARING));  //FIXME: is bearing equal to rotate in baidu map?
+		camera.overlook(settings.getFloat(PREF_TILT, DEFAULT_TILT));  //FIXME: is tilt equal to overlook in baidu map?
+		camera.zoom(settings.getFloat(PREF_ZOOM, DEFAULT_ZOOM_LEVEL));
+		camera.target(new LatLng(settings.getFloat(PREF_LAT, DEFAULT_LATITUDE),
+                settings.getFloat(PREF_LNG, DEFAULT_LONGITUDE)));
+
+		map.setMapStatus(MapStatusUpdateFactory.newMapStatus(camera.build()));
+    }
+
+    private void setupMap() {
+        final BaiduMap map = getBaiduMap();
+		if (map == null) return;
+
+  		setupMapUI(map);
+		setupMapOverlay(map);
+		setupMapListeners(map);
+    }
+
+    @Override
+    public void zoomToFit(List<LatLong> coords) {
+        final BaiduMap map = getBaiduMap();
+		if (map == null) return;
+
+		if (!coords.isEmpty()) {
+            final List<LatLng> points = new ArrayList<LatLng>();
+            for (LatLong coord : coords)
+                points.add(CoordToLatLang(coord));
+
+            final LatLngBounds bounds = getBounds(points);
+
+			final Activity activity = getActivity();
+			if(activity == null)
+				return;
+
+			final View rootView = ((ViewGroup)activity.findViewById(android.R.id.content)).getChildAt(0);
+			if(rootView == null)
+				return;
+
+			final int height = rootView.getHeight();
+			final int width = rootView.getWidth();
+			Log.d(TAG, String.format(Locale.US, "Screen W %d, H %d", width, height));
+			if(height > 0 && width > 0) {
+				MapStatusUpdate animation = MapStatusUpdateFactory.newLatLngBounds(bounds, width, height); //, 100); FIXME: What does 100 mean? zoom?
```

Comment:
@droneboost in the Google map implementation, `100` is the amount of padding in pixels. It could be the same here too.
But you're right, it could be stored into a constant to give a better indication as to its use.